src/Illuminate/Filesystem/functions.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace Illuminate\Filesystem;
+
+if (! function_exists('Illuminate\Filesystem\join_paths')) {
+    /**
+     * Join the given paths together.
+     *
+     * @param  string  $basePath
+     * @param  string  $path
+     * @return string
+     */
+    function join_paths($basePath, $path = '')
+    {
+        return $basePath.($path != '' ? DIRECTORY_SEPARATOR.ltrim($path, DIRECTORY_SEPARATOR) : '');
+    }
```

Comment:
I'd love it this just accepted an arbitrary list of paths, like Node's version.

```php
join_paths('storage', 'keys', 'public', 'key.txt')
```

Reply:
Done.